src/Host/Client/Test/Mocks/RSessionMock.cs

```diff
@@ -40,17 +40,17 @@ public sealed class RSessionMock : IRSession {
 
         public Task<IRSessionInteraction> BeginInteractionAsync(bool isVisible = true, CancellationToken cancellationToken = default (CancellationToken)) {
             _inter = new RSessionInteractionMock();
-            BeforeRequest?.Invoke(this, new RRequestEventArgs(_inter.Contexts, Prompt, 4096, true));
+            BeforeRequest?.Invoke(this, new RBeforeRequestEventArgs(_inter.Contexts, Prompt, 4096, true));
             return Task.FromResult(_inter);
         }
 
         public Task CancelAllAsync() {
```

Comment:
No, this is Mock implementation, real implementation needs to be async

---
In reply to: [71068173](https://github.com/Microsoft/RTVS/pull/2080#discussion_r71068173) [](ancestors = 71068173)